README.md
# mappings-collaboratory

> List of ID/string mappings for collaboratory

This reference lib is used to help translate ID numbers into human-friendly strings.
(and vice versa) It does so by exposing several properties on the exported object,
where each one has a `by_id` and `by_name` plain JS object.

The `by_id` object gives the human-friendly string from just an ID number.
(eg: `roles.by_id[0]`)

The `by_name` object gives an ID number from a string. We use
[to-constant-case](https://github.com/ianstormtaylor/to-constant-case) to normalize
the string name, so things like `"Citizen"` become `"CITIZEN"` while more complex names
like `"Non-Profit"` become `"NON_PROFIT"`.

## Usage

```js
var mapping = require("mappings-collaboratory");

console.log(mapping.roles.by_name.MEMBER);
// => 5

console.log(mapping.roles.by_id[5]);
// => "Member"
```


CHANGES.md
# 0.0.25 (08/16/2017)
 * Added new collaborator type

# 0.0.24 (07/03/2017)
 * Added new group type

# 0.0.23 (04/18/2017)
 * Added collaborator types

# 0.0.22 (02/21/2017)
 * Fixing organization typo

# 0.0.21 (02/21/2017)
 * Update organization types

# 0.0.20 (01/06/2017)
 * Update unit types

 # 0.0.19 (01/04/2017)
 * Update faculty/staff position types

# 0.0.18 (12/06/2016)
 * Add staff position types

# 0.0.17 (10/13/2016)
 * Update relationship types

# 0.0.16 (09/27/2016)
 * Add "Museum" and "Clinic" to group types

# 0.0.15 (06/14/2016)
 * Add Back in Edge Types to prevent breaking indexes

# 0.0.14 (06/06/2016)
 * Add New Edge Types

# 0.0.13 (02/23/2016)
 * Add Vertex Type ID for Partnership

# 0.0.12 (02/08/2016)
 * Update string to prevent constant case from adding underscores between chars

# 0.0.11 (01/21/2016)
 * Update dependency to use treetopllc fork

# 0.0.10 (10/12/2015)
 * Update 'Citizen' verbiage to 'Member' on role types

# 0.0.9 (8/03/2015)
 * Add properties to statuses

# 0.0.8 (2/28/2015)
 * Change "Citizen" to "Member"

# 0.0.7 (2/28/2015)
 * update Collaboratory specific group types

# 0.0.6 (6/18/2015)
 * Add Division Group Type

# 0.0.5 (5/13/2015)
 * Add vertex type ID for Partnership

# 0.0.4 (4/28/2015)
 * Update Project Vertex Type Name to Activity

# 0.0.3 (4/27/2015)
 * Add vertex and relationship types

# 0.0.2 (9/5/2014)
 * Add asset types

# 0.0.1 (9/5/2014)
 * initial release (cloned from treetopllc/mappings-nobleweb)


index.js
var constant = require("to-constant-case");
var each = require("each");

exports.edge_types = index({
    0:  "Parent",
    1:  "Content",
    2:  "Admin",
    3:  "Moderator",
    4:  "Contributor",
    5:  "Member",
    6:  "Follower",
    7:  "Like",
    8:  "Author",
    9:  "Verifier",
    10: "Submission",
    11: "Hours",
    12: "Revision",
    13: "Location",
    14: "Email",
    15: "Partner",
    16: "Proxy",
    17: "Relationship",
    18: "Transfer Admin",
    19: "Host"
});

exports.event_types = index([
    "Celebration",
    "Culinary Arts",
    "Demonstrations",
    "Discussions",
    "Educational",
    "Family",
    "Films",
    "Fundraising",
    "Just for Fun",
    "Lecture",
    "Performance",
    "Workshop",
    "(Other)",
    "Conferences"
]);

exports.group_types = index({
    3: "Course",
    4: "Academic Department/Program",
    5: "Office",
    6: "Center/Institute",
    7: "Division",
    8: "Museum",
    9: "Clinic",
    10: "Network/Coalition",
    11: "Co-Curricular"
});

exports.opportunity_types = index({
    "-1": "Offline",
    0: "Internship",
    1: "Paying Job",
    2: "Volunteer Service",
    3: "Mentoring",
    4: "Tutoring",
    5: "Service Learning",
    6: "Performance"
});

exports.organization_types = index({
    "-1": "Offline",
    0: "Community Group",
    1: "Educational Institution",
    2: "For-profit",
    3: "Non-profit",
    4: "Government",
    5: "Social/Civic",
    6: "Youth Association",
    7: "Unaffiliated Individual"
});

exports.activity_types = index([
    "Teaching",
    "Research/Creative",
    "Service"
]);

exports.activity_focuses = index([
    "Arts and Culture",
    "Education",
    "Community and Economic Development",
    "Government/Public Safety",
    "Environment and Sustainability",
    "Health and Wellness",
    "Social Issues"
]);

exports.roles = index({
    1: "Noblehour Admin",
    2: "Admin",
    3: "Moderator",
    4: "Contributor",
    5: "Member",
    6: "Follower"
});

exports.submission_statuses = index({
    0: "Unsubmitted",
    1: "Approved",
    2: "Denied",
    3: "Pending"
});

exports.vertex_types = index({
    0: "News",
    1: "Group",
    2: "Organization",
    3: "Opportunity",
    4: "Activity",
    5: "Event",
    6: "Customer",
    7: "User",
    8: "Submission",
    9: "Submission History",
    10: "Hours",
    11: "Address",
    12: "Asset",
    13: "Url",
    14: "Hours Set",
    15: "Email",
    16: "Relationship",
    17: "Goal",
    18: "Invite",
    19: "Proxy",
    20: "Program",
    21: "Partnership"
});

exports.activity_types = index([
    "Create",
    "Update",
    "Delete"
]);

exports.asset_types = index({
    0: "notype",
    1: "aif",
    2: "doc",
    3: "docx",
    4: "gif",
    5: "jpg",
    6: "mov",
    7: "mp3",
    8: "mpg",
    9: "odp",
    10: "odt",
    11: "oga",
    12: "ogf",
    13: "pdf",
    14: "png",
    15: "pps",
    16: "ppsx",
    17: "ppt",
    18: "pptx",
    19: "txt",
    20: "wma",
    21: "wmv",
    22: "xls",
    23: "xlsx",
    24: "rtf"
});

exports.relationship_types = index({
    1: "Student",
    2: "Faculty",
    3: "Staff",
    8: "Assistant professor",
    9: "Associate professor",
    10: "Professor",
    11: "Other instructional role",
    12: "Clinician",
    14: "Adjunct instructor",
    16: "Profile Holder",
    17: "Executive leadership",
    18: "Leader of a center, office, or institute",
    19: "Support for units or programs"
});

exports.collaborator_types = index({
    "-1": "None",
    0: "Community Partner",
    1: "Institutional Partner",
    2: "Campus Partner",
    3: "Faculty/Staff",
    4: "Course",
    5: "Program"
});

function index(list) {
    var by_name = {};
    var by_id = {};

    each(list, !!list.length ? array : add);

    function array(name, id) {
        add(id, name);
    }

    function add(id, name) {
        by_id[id] = name;
        by_name[constant(name)] = parseInt(id, 10);
    }

    return {
        by_id: by_id,
        by_name: by_name
    };
}
